Makefile

```diff
@@ -349,115 +356,53 @@ $(BATS):
-	$(BATS) --trace -p --verbose-run --print-output-on-failure --show-output-of-passing-tests test/blackbox/scrub.bats
-
-.PHONY: test-bats-metrics
-test-bats-metrics: BUILD_LABELS=metrics
-test-bats-metrics: check-linux binary check-skopeo $(BATS)
-	$(BATS) --trace --print-output-on-failure test/blackbox/metrics.bats
-
-.PHONY: test-bats-metrics-verbose
-test-bats-metrics-verbose: BUILD_LABELS=metrics
-test-bats-metrics-verbose: check-linux binary check-skopeo $(BATS)
-	$(BATS) --trace -p --verbose-run --print-output-on-failure --show-output-of-passing-tests test/blackbox/metrics.bats
-
-.PHONY: test-anonymous-push-pull
-test-anonymous-push-pull: check-linux binary check-skopeo $(BATS)
-	$(BATS) --trace --print-output-on-failure test/blackbox/anonymous_policy.bats
-
-.PHONY: test-annotations
-test-annotations: check-linux binary check-skopeo $(BATS) $(STACKER) $(NOTATION) $(COSIGN)
-	$(BATS) --trace --print-output-on-failure test/blackbox/annotations.bats
-
-.PHONY: test-detect-manifest-collision
-test-detect-manifest-collision: check-linux binary check-skopeo $(BATS) $(REGCLIENT)
-	$(BATS) --trace --print-output-on-failure test/blackbox/detect_manifest_collision.bats
+.PHONY: check-blackbox-prerequisites
+check-blackbox-prerequisites: check-linux check-skopeo $(BATS) $(REGCLIENT) $(ORAS) $(HELM) $(CRICTL) $(NOTATION) $(COSIGN)
+	which skopeo; \
+	which regctl; \
+	which oras; \
+	which helm; \
+	which crictl; \
+	which notation; \
+	which cosign
+
+.PHONY: run-blackbox-tests
+run-blackbox-tests: $(BATS_TEST_FILE_PATH) check-blackbox-prerequisites binary binary-minimal bench
+	echo running bats test "$(BATS_TEST_FILE_PATH)"; \
+	$(BATS) $(BATS_FLAGS) $(BATS_TEST_FILE_PATH)
+
+.PHONY: run-blackbox-ci
+run-blackbox-ci: check-blackbox-prerequisites binary binary-minimal
+	echo running CI bats tests; \
+	$(BATS) $(BATS_FLAGS) test/blackbox/pushpull.bats; \
+	$(BATS) $(BATS_FLAGS) test/blackbox/delete_images.bats; \
+	$(BATS) $(BATS_FLAGS) test/blackbox/referrers.bats; \
+	$(BATS) $(BATS_FLAGS) test/blackbox/metadata.bats; \
+	$(BATS) $(BATS_FLAGS) test/blackbox/anonymous_policy.bats; \
+	$(BATS) $(BATS_FLAGS) test/blackbox/annotations.bats; \
+	$(BATS) $(BATS_FLAGS) test/blackbox/detect_manifest_collision.bats; \
+	$(BATS) $(BATS_FLAGS) test/blackbox/cve.bats; \
+	$(BATS) $(BATS_FLAGS) test/blackbox/sync.bats; \
+	$(BATS) $(BATS_FLAGS) test/blackbox/sync_docker.bats; \
+	$(BATS) $(BATS_FLAGS) test/blackbox/sync_replica_cluster.bats; \
+	$(BATS) $(BATS_FLAGS) test/blackbox/scrub.bats; \
+	$(BATS) $(BATS_FLAGS) test/blackbox/metrics.bats
+
+.PHONY: run-blackbox-cloud-ci
+run-blackbox-cloud-ci: check-blackbox-prerequisites binary $(BATS)
+	echo running cloud CI bats tests; \
+	$(BATS) $(BATS_FLAGS) test/blackbox/cloud-only.bats
+
+.PHONY: run-blackbox-dedupe-nightly
+run-blackbox-dedupe-nightly: check-blackbox-prerequisites binary binary-minimal
+	echo running nightly dedupe tests; \
+	$(BATS) $(BATS_FLAGS) test/blackbox/restore_s3_blobs.bats; \
```

Comment:
If this command fails, and the next one succeeds, the whole target will succeeds.
same for CI target.

```
midgard@yggdrasil:~/work/github.com/andaaron/zot$ make run-blackbox-dedupe-nightly 
skopeo -v || (echo "You need skopeo to be installed in order to run tests"; exit 1)
skopeo version 1.13.1
which skopeo; \
which regctl; \
which oras; \
which helm; \
which crictl; \
which notation; \
which cosign
/usr/bin/skopeo
/home/midgard/work/github.com/andaaron/zot/hack/tools/bin/regctl
/home/midgard/work/github.com/andaaron/zot/hack/tools/bin/oras
/home/midgard/work/github.com/andaaron/zot/hack/tools/bin/helm
/home/midgard/work/github.com/andaaron/zot/hack/tools/bin/crictl
/home/midgard/work/github.com/andaaron/zot/hack/tools/bin/notation
/home/midgard/work/github.com/andaaron/zot/hack/tools/bin/cosign
go mod tidy
echo imagetrust,lint,metrics,mgmt,scrub,search,sync,ui,userprefs;\
if [ -z commit-b787273 ]; then\
	pwd=$(pwd);\
	tdir=$(mktemp -d);\
	cd $tdir;\
	git clone https://github.com/project-zot/zui.git;\
	cd zui;\
	npm install;\
	npm run build;\
	cd $pwd;\
	rm -rf ./pkg/extensions/build;\
	cp -R $tdir/zui/build ./pkg/extensions/;\
else\
	curl --fail --head https://github.com/project-zot/zui/releases/download/commit-b787273/zui.tgz;\
	if [ $? -ne 0 ]; then\
		pwd=$(pwd);\
		tdir=$(mktemp -d);\
		cd $tdir;\
  			git clone --depth=1 --branch commit-b787273  https://github.com/project-zot/zui.git;\
		cd zui;\
		git checkout commit-b787273;\
		npm install;\
		npm run build;\
		cd $pwd;\
		rm -rf ./pkg/extensions/build;\
		cp -R $tdir/zui/build ./pkg/extensions/;\
	else\
 			curl -fsSL https://github.com/project-zot/zui/releases/download/commit-b787273/zui.tgz -o zui.tgz;\
		tar xvzf zui.tgz -C ./pkg/extensions/;\
		rm zui.tgz;\
	fi;\
fi;\

imagetrust,lint,metrics,mgmt,scrub,search,sync,ui,userprefs
HTTP/2 302 
server: GitHub.com
date: Fri, 01 Sep 2023 16:07:41 GMT
content-type: text/html; charset=utf-8

..........................



env CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -o bin/zot-linux-amd64-minimal -buildmode=pie -tags containers_image_openpgp -v -trimpath -ldflags "-X zotregistry.io/zot/pkg/api/config.ReleaseTag=v1.1.13 -X zotregistry.io/zot/pkg/api/config.Commit=v1.1.13-692-g925e751 -X zotregistry.io/zot/pkg/api/config.BinaryType=minimal -X zotregistry.io/zot/pkg/api/config.GoVersion=go1.21.0 -s -w" ./cmd/zot
echo running nightly dedupe tests; \
/home/midgard/work/github.com/andaaron/zot/hack/tools/bin/bats --trace --print-output-on-failure test/blackbox/restore_s3_blobs.bats; \
/home/midgard/work/github.com/andaaron/zot/hack/tools/bin/bats --trace --print-output-on-failure test/blackbox/pushpull_running_dedupe.bats
running nightly dedupe tests
restore_s3_blobs.bats
 ✗ setup_file failed
you need to install aws cli as a prerequisite to running the tests
   (from function `teardown_file' in test file test/blackbox/restore_s3_blobs.bats, line 92)
     `awslocal s3 rb s3://"zot-storage" --force' failed with status 0
   /home/midgard/work/github.com/andaaron/zot/test/blackbox/restore_s3_blobs.bats: line 92: awslocal: command not found
   /home/midgard/work/github.com/andaaron/zot/test/blackbox/restore_s3_blobs.bats: line 93: awslocal: command not found
   bats warning: Executed 1 instead of expected 3 tests

3 tests, 1 failure, 2 not run

pushpull_running_dedupe.bats
 ✓ push image - dedupe not running
push image exec time: 2 sec
 ✓ pull image - dedupe not running
pull image exec time: 1 sec
 ✓ push 50 images with dedupe disabled
 ✓ restart zot with dedupe enabled
 ✓ push image - dedupe running
push image exec time: 0 sec
 ✓ pull image - dedupe running
pull image exec time: 1 sec
 ✓ pull deduped image - dedupe running
pull image exec time: 1 sec
 ✓ push image index - dedupe running
push image index exec time: 15 sec
 ✓ pull image index - dedupe running
pull image index exec time: 0 sec
 ✓ push oras artifact - dedupe running
push oras artifact exec time: 0 sec
 ✓ pull oras artifact - dedupe running
pull oras artifact exec time: 0 sec
 ✓ attach oras artifacts - dedupe running
attach signature exec time: 0 sec
attach sbom exec time: 0 sec
 ✓ discover oras artifacts - dedupe running
discover oras artifacts exec time: 0 sec
 ✓ push helm chart - dedupe running
helm push exec time: 0 sec
 ✓ pull helm chart - dedupe running
helm pull exec time: 0 sec
 ✓ push image with regclient - dedupe running
regclient push exec time: 0

16 tests, 0 failures

midgard@yggdrasil:~/work/github.com/andaaron/zot$ $?
0: command not found
```